### PR TITLE
Multiple submit event(s) prevention logic in recovery portals

### PR DESCRIPTION
**Pull request by yasinmiran**
## Purpose
Resolves https://github.com/wso2/product-is/issues/10924

asdjasdad

### CODE REVIEW

File: apps/recovery-portal/src/main/webapp/password-recovery.jsp [hunk 1]
```diff
@@ -264,6 +264,12 @@
         $(document).ready(function () {
 
             $("#recoverDetailsForm").submit(function (e) {
+
+                // Prevent clicking multiple times, and notify the user something
+                // is happening in the background.
+                const submitButton = $('#recoverySubmit');
```
Comment (DimalChandrasiri) [inline apps/recovery-portal/src/main/webapp/password-recovery.jsp:270]:
```suggestion
                const submitButton = $("#recoverySubmit");
```

Comment (yasinmiran) [reply apps/recovery-portal/src/main/webapp/password-recovery.jsp]: Fixed the linting issue in the commit a56cb5d. Thanks! :smile:

File: apps/recovery-portal/src/main/webapp/password-recovery.jsp [hunk 2]
```diff
@@ -275,11 +281,11 @@
                 // Validate User Name
                 var firstName = $("#username").val();
 
-                if (firstName == '') {
+                if (firstName === '') {
```
Comment (DimalChandrasiri) [inline apps/recovery-portal/src/main/webapp/password-recovery.jsp]:
```suggestion
                if (firstName === "") {
```

File: apps/recovery-portal/src/main/webapp/password-recovery.jsp [hunk 3]
```diff
@@ -288,11 +294,11 @@
 
                 var reCaptchaResponse = $("[name='g-recaptcha-response']")[0].value;
 
-                if (reCaptchaResponse.trim() == '') {
+                if (reCaptchaResponse.trim() === '') {
```
Comment (DimalChandrasiri) [inline apps/recovery-portal/src/main/webapp/password-recovery.jsp:297]:
```suggestion
                if (reCaptchaResponse.trim() === "") {
```

File: apps/recovery-portal/src/main/webapp/username-recovery.jsp [hunk 4]
```diff
@@ -316,17 +316,23 @@
         $(document).ready(function () {
 
             $("#recoverDetailsForm").submit(function (e) {
+
+                // Prevent clicking multiple times, and notify the user something
+                // is happening in the background.
+                const submitButton = $('#recoverySubmit');
```
Comment (DimalChandrasiri) [inline apps/recovery-portal/src/main/webapp/username-recovery.jsp]:
```suggestion
                const submitButton = $("#recoverySubmit");
```

File: apps/recovery-portal/src/main/webapp/username-recovery.jsp [hunk 5]
```diff
@@ -316,17 +316,23 @@
...
+                submitButton.addClass("loading").attr("disabled", true);
+
                 var errorMessage = $("#error-msg");
                 errorMessage.hide();
 
                 <% if (isFirstNameInClaims){ %>
                     var firstName = $("#first-name").val();
 
-                    if (firstName == '') {
+                    if (firstName === '') {
```
Comment (DimalChandrasiri) [inline apps/recovery-portal/src/main/webapp/username-recovery.jsp]:
```suggestion
                    if (firstName === "") {
```

**Review by DimalChandrasiri — APPROVED** [review]

File: apps/recovery-portal/src/main/webapp/username-recovery.jsp (hunk 5 above)
Comment (yasinmiran) [reply apps/recovery-portal/src/main/webapp/username-recovery.jsp]: Fixed in commit f85c0f0

File: apps/recovery-portal/src/main/webapp/username-recovery.jsp (hunk 4 above)
Comment (yasinmiran) [reply apps/recovery-portal/src/main/webapp/username-recovery.jsp]: Fixed in commit f85c0f0

File: apps/recovery-portal/src/main/webapp/password-recovery.jsp (hunk 3 above)
Comment (yasinmiran) [reply apps/recovery-portal/src/main/webapp/password-recovery.jsp]: Fixed the equality check, in commit f85c0f0

File: apps/recovery-portal/src/main/webapp/password-recovery.jsp (hunk 2 above)
Comment (yasinmiran) [reply apps/recovery-portal/src/main/webapp/password-recovery.jsp]: Fixed in commit f85c0f0

**Pull request merged**